src/ert/shared/models/base_run_model.py

```diff
@@ -430,38 +433,31 @@ def id_(self, value: str) -> None:
     async def _ensemble_listener(self, ensemble: Ensemble, iter_: int) -> None:
         """Redirect events emitted by the ensemble to this experiment."""
         while True:
-            event: CloudEvent = await ensemble.output_bus.get()
-            await self.dispatch(event, iter_)
-            if event["type"] in (
-                identifiers.EVTYPE_ENSEMBLE_FAILED,
-                identifiers.EVTYPE_ENSEMBLE_CANCELLED,
-                identifiers.EVTYPE_ENSEMBLE_STOPPED,
-            ):
-                break
-
-    async def _dispatch_ee(
-        self, ee_event_type: str, iteration: int, data: Optional[dict] = None
+            event: com_protocol.DispatcherMessage = await ensemble.output_bus.get()
+            await self.dispatch(event)
+            if event.WhichOneof("object") == "ensemble":
```

Comment:
good suggestion by "annotate-python-linting", use a single if-statement instead of nested.